server/lib/chat/chat-service.ts

```diff
@@ -217,27 +206,40 @@ export default class ChatService {
         "Can't moderate yourself",
       )
     }
+
+    const { channelPermissions: moderatorChannelPermissions } = moderatorUserChannel
+    const { channelPermissions: targetChannelPermissions } = targetUserChannel
+
+    // TODO(2Pac): Really need tests for these.
+
+    if (targetChannelPermissions.owner && !moderatorPermissions?.moderateChatChannels) {
```

Comment:
imo you should really name these associated conditions by putting them in a variable, and then using those variables in these if statements. These are *absurdly* hard to read and comprehend what the underlying intention is meant to be (and sure, tests would help ensure they don't break in the future, but it would be better to also write code we can actually understand)